tests/compendia_tests.py

```
@@ -90,46 +98,54 @@ class CompendiaTests(unittest.TestCase):
-        self.assertEqual(actual.computed_file.is_qc, expected["computed_file"]["is_qc"])
-        self.assertEqual(actual.computed_file.sha1, expected["computed_file"]["sha1"])
-        self.assertEqual(actual.computed_file.s3_bucket, expected["computed_file"]["s3_bucket"])
-        self.assertEqual(actual.computed_file.s3_key, expected["computed_file"]["s3_key"])
-        self.assertEqual(actual.computed_file.created_at, expected["computed_file"]["created_at"])
-        self.assertEqual(actual.computed_file.last_modified, expected["computed_file"]["last_modified"])
-
-        # Assert that non-returned feilds are set to their default values
-        self.assertIsNone(actual.computed_file.compendia_organism_name)
-        self.assertIsNone(actual.computed_file.compendia_version)
-        self.assertIsNone(actual.computed_file.download_url)
-        self.assertIsNone(actual.computed_file.quant_sf_only)
-        self.assertIsNone(actual.computed_file.result)
-        self.assertIsNone(actual.computed_file.s3_url)
-        self.assertListEqual(actual.computed_file.samples, [])
+        result_list = list(result)
+
+        self.assertObject(result_list[0], compendia_object_1)
+        self.assertObject(result_list[1], compendia_object_2)
+
+        self.assertEqual(len(mock_get.call_args_list), 2)
 
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_search_with_invalid_filters(self, mock_get):
+        result = pyrefinebio.Compendia.search(foo="bar")
+
+        result_list = list(result)
+
+        self.assertObject(result_list[0], compendia_object_1)
+        self.assertObject(result_list[1], compendia_object_2)
+
+        self.assertEqual(len(mock_get.call_args_list), 2)
+
+    @patch("pyrefinebio.http.requests.get", side_effect=mock_get_request)
+    def test_compendia_500(self, mock_get):
+        with self.assertRaises(Exception):
```

We should return a meaningful exception type with a descriptive/helpful message.